src/daemon/daemon_init_settings.cpp

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (C) 2017-2022 Canonical, Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "daemon_init_settings.h"
+
+#include <multipass/constants.h>
+#include <multipass/platform.h>
+#include <multipass/settings/basic_setting_spec.h>
+#include <multipass/settings/bool_setting_spec.h>
+#include <multipass/settings/dynamic_setting_spec.h>
+#include <multipass/settings/persistent_settings_handler.h>
+#include <multipass/settings/settings.h>
+#include <multipass/utils.h>
+
+#include <QCoreApplication>
+#include <QFileSystemWatcher>
+#include <QObject>
+
+namespace mp = multipass;
+
+namespace
+{
+constexpr const int settings_changed_code = 42;
+
+/*
+ * We make up our own file names to:
+ *   a) avoid unknown org/domain in path;
+ *   b) write daemon config to a central location (rather than user-dependent)
+ * Example: /root/.config/multipassd/multipassd.conf
+ */
+QString persistent_settings_filename()
+{
+    static const auto file_pattern = QStringLiteral("%2%1").arg(mp::settings_extension);
+    static const auto daemon_dir_path = QDir{MP_PLATFORM.daemon_config_home()}; // temporary, replace w/ AppConfigLoc
+    static const auto path = daemon_dir_path.absoluteFilePath(file_pattern.arg(mp::daemon_name));
+
+    return path;
+}
+
+QString driver_interpreter(QString val)
+{
+    if (!MP_PLATFORM.is_backend_supported(val))
+        throw mp::InvalidSettingException(mp::driver_key, val, "Invalid driver");
+
+    return val;
+}
+
+} // namespace
+
+void mp::daemon::monitor_and_quit_on_settings_change() // temporary
+{
+    static const auto filename = persistent_settings_filename();
+    mp::utils::check_and_create_config_file(filename); // create if not there
+
+    static QFileSystemWatcher monitor{{filename}};
+    QObject::connect(&monitor, &QFileSystemWatcher::fileChanged, [] { QCoreApplication::exit(settings_changed_code); });
+}
```

Comment:
Do we still need this at all? The only time we want to exit is on driver change, so couldn't a settings handler for the driver issue the exit?

Reply:
I initially thought I was going to remove it, but I would need to replace it with something similar. 

If the handler exited right away, it would cut the line back to the client, which would see that as an error (rightfully so, I think, as it couldn't know that the RPC setting was successful). So I would need some other asynchronous mechanism for delayed reaction. It turns out that this provides just such a mechanism, as it will only be processed when we next enter the event loop.